ic10emu/src/device.rs

```diff
@@ -549,10 +546,10 @@ pub struct Device {
     pub name_hash: Option<i32>,
     pub prefab: Option<Prefab>,
     pub slots: Vec<Slot>,
-    pub reagents: HashMap<ReagentMode, HashMap<i32, f64>>,
+    pub reagents: BTreeMap<ReagentMode, BTreeMap<i32, f64>>,
     pub ic: Option<u32>,
     pub connections: Vec<Connection>,
-    fields: HashMap<LogicType, LogicField>,
+    fields: BTreeMap<LogicType, LogicField>,
```

Comment:
nice, also noticed this

Reply:
As I noted in the commit comment I decided stable field order was necessary. it means less front-end updates and stable serialising